internal/engine/interpreter/interpreter.go

```diff
@@ -277,85 +277,15 @@ func (e *engine) lowerIR(ir *wazeroir.CompilationResult) (*code, error) {
 	labelAddress := map[wazeroir.LabelID]uint64{}
 	onLabelAddressResolved := map[wazeroir.LabelID][]func(addr uint64){}
 	for i, original := range ops {
-		var op *wazeroir.UnionOperation
-		if o, ok := original.(wazeroir.UnionOperation); ok {
-			op = &o
-		} else {
-			op = &wazeroir.UnionOperation{OpKind: original.Kind()}
-		}
+		// copy to op to avoid mutation in-place
+		op := original
```

Comment:
do we really need this?

Reply:
If I remove that, several tests start to fail (e.g. `TestInterpreter_ModuleEngine_Call_HostFn`). Maybe there is a better way?

Reply:
instead of doing a copy I am now doing `op := &ops[i]`; this works but I wonder if the mutation at `op.SourcePC =...` is ok...